Source/WebCore/dom/Element.cpp

```diff
@@ -2033,7 +2041,36 @@ void Element::setElementAttribute(const QualifiedName& attributeName, Element* e
     explicitlySetAttrElementsMap().set(attributeName, Vector<WeakPtr<Element>> { element });
 }
 
-std::optional<Vector<RefPtr<Element>>> Element::getElementsArrayAttribute(const QualifiedName& attributeName) const
+JSC::JSValue Element::getElementsArrayAttribute(const QualifiedName& attributeName, JSC::JSGlobalObject& lexicalGlobalObject, WebCore::JSDOMGlobalObject& globalObject)
+{
+    JSValueInWrappedObject* cachedValue = nullptr;
+    if (auto* cachedMap = cachedAttrAssociatedElementsMapIfExists()) {
+        if (auto it = cachedMap->find(attributeName); it != cachedMap->end())
+            cachedValue = it->value.get();
+    }
+
+    std::optional<Vector<RefPtr<Element>>> elements = getElementsArrayAttributeInternal(attributeName);
+    if (!elements) {
+        if (cachedValue)
+            cachedValue->clear();
+        return JSC::jsNull();
+    }
+
+    JSC::JSValue newValue = JSConverter<IDLFrozenArray<IDLInterface<Element>>>::convert(lexicalGlobalObject, globalObject, elements.value());
+    if (cachedValue) {
+        if (!cachedValue->getValue().isUndefinedOrNull()) {
+            Vector<RefPtr<Element>> cachedElements = Converter<IDLFrozenArray<IDLInterface<Element>>>::convert(lexicalGlobalObject, cachedValue->getValue());
+            if (elements == cachedElements)
+                return cachedValue->getValue();
+        }
+
+        cachedValue->setWeakly(newValue);
+    } else
+        ensureElementRareData().cachedAttrAssociatedElementsMap().set(attributeName, makeUnique<JSValueInWrappedObject>(newValue));
```

Comment:
I'm pretty sure this is not thread safe. r-

Reply:
You likely want to use `cachedPropertyValue()` from JSValueInWrappedObject.h. See existing examples in the code base. `cachedPropertyValue()` has the right DOM wrapper world checks.